src/app/clusters/groups-server/groups-server.cpp

```diff
@@ -188,38 +239,34 @@ bool emberAfGroupsClusterAddGroupCallback(EndpointId endpoint, app::CommandHandl
     return true;
 }
 
-bool emberAfGroupsClusterViewGroupCallback(EndpointId endpoint, app::CommandHandler * commandObj, GroupId groupId)
+/**
+ * @brief  Cluster ViewGroup Command callback (from client)
+ */
+bool emberAfGroupsClusterViewGroupCallback(chip::EndpointId endpoint, chip::app::CommandHandler * commandObj, uint16_t groupId)
 {
-    EmberAfStatus status                                          = EMBER_ZCL_STATUS_NOT_FOUND;
-    CHIP_ERROR err                                                = CHIP_NO_ERROR;
-    uint8_t groupName[ZCL_GROUPS_CLUSTER_MAXIMUM_NAME_LENGTH + 1] = { 0 };
-
-    // Get the group name, if supported
-    emberAfPluginGroupsServerGetGroupNameCallback(emberAfCurrentEndpoint(), groupId, groupName);
-
-    emberAfGroupsClusterPrintln("RX: ViewGroup 0x%2x", groupId);
+    CHIP_ERROR err = CHIP_NO_ERROR;
 
-    // For all networks, View Group commands can only be addressed to a
-    // single device.
-    if (emberAfCurrentCommand()->type != EMBER_INCOMING_UNICAST && emberAfCurrentCommand()->type != EMBER_INCOMING_UNICAST_REPLY)
+    // For all networks, View Group commands can only be addressed to a single device.
+    if (emberAfCurrentCommand()->type == EMBER_INCOMING_UNICAST || emberAfCurrentCommand()->type == EMBER_INCOMING_UNICAST_REPLY)
     {
-        return true;
-    }
+        uint8_t groupName[ZCL_GROUPS_CLUSTER_MAXIMUM_NAME_LENGTH + 1] = { 0 };
 
-    if (isGroupPresent(emberAfCurrentEndpoint(), groupId))
-    {
-        status = EMBER_ZCL_STATUS_SUCCESS;
-    }
+        EmberAfStatus status = checkGroup(endpoint, groupId);
+        if (EMBER_ZCL_STATUS_SUCCESS == status)
+        {
+            // Get the group name, if supported
+            emberAfPluginGroupsServerGetGroupNameCallback(endpoint, groupId, groupName);
```

Comment:
Since `emberAfPluginGroupsServerGetGroupNameCallback` does nothing, we cannot assume `groupName` will be properly populated. How do we error-check this?